bundles/binding/org.openhab.binding.primare/src/main/java/org/openhab/binding/primare/internal/PrimareBinding.java

```diff
@@ -0,0 +1,478 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.primare.internal;
+
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.EventObject;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openhab.binding.primare.PrimareBindingProvider;
+import org.openhab.binding.primare.internal.protocol.PrimareConnector;
+import org.openhab.binding.primare.internal.protocol.PrimareSerialConnector;
+import org.openhab.binding.primare.internal.protocol.PrimareTCPConnector;
+import org.openhab.binding.primare.internal.protocol.PrimareUtils;
+import org.openhab.binding.primare.internal.protocol.PrimareResponse;
+import org.openhab.core.binding.AbstractBinding;
+import org.openhab.core.binding.BindingChangeListener;
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.DimmerItem;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.RollershutterItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+	
+
+/**
+ * Binding listening to openHAB event bus and send commands to Primare devices when certain
+ * commands are received.
+ * 
+ * @author Pauli Anttila, Veli-Pekka Juslin
+ * @since 1.7.0
+ */
+public class PrimareBinding extends AbstractBinding<PrimareBindingProvider> 
+	implements ManagedService, BindingChangeListener, PrimareEventListener {
+
+	private static final Logger logger = LoggerFactory.getLogger(PrimareBinding.class);
+
+	protected static final String WILDCARD_COMMAND_KEY = "*";
+
+	/** RegEx to validate a config <code>'^(.*?)\\.(host|port|serial|model)$'</code> */
+	private static final Pattern EXTRACT_CONFIG_PATTERN = Pattern.compile("^(.*?)\\.(host|port|serial|model)$");
+	
+	/** Map table to store all available receivers configured by the user */
+	protected Map<String, DeviceConfig> deviceConfigCache = null;
+	
+	public PrimareBinding() {
+	}
+
+	public void activate() {
+		logger.debug("Activate");
+	}
+
+	public void deactivate() {
+		logger.debug("Deactivate");
```

Comment:
Could you please elaborate?

Should the following be removed:

1. The constructor which is effectively the default constructor:

public PrimareBinding() {
}


2. activate()

public void activate() {
   logger.debug("Activate");
}


3. deactivate()

public void deactivate() {
   logger.debug("Deactivate");
   closeAllConnections();
}


Or do you mean just the logger.debug() statements?


/vp



On 19.5.2015 22:05, Thomas Eichstädt-Engelen wrote:
> In
> bundles/binding/org.openhab.binding.primare/src/main/java/org/openhab/binding/primare/internal/PrimareBinding.java
> <https://github.com/openhab/openhab/pull/2263#discussion_r30634708>:
>
>> +
>> +	/** RegEx to validate a config <code>'^(.*?)\\.(host|port|serial|model)$'</code> */
>> +	private static final Pattern EXTRACT_CONFIG_PATTERN = Pattern.compile("^(.*?)\\.(host|port|serial|model)$");
>> +	
>> +	/** Map table to store all available receivers configured by the user */
>> +	protected Map<String, DeviceConfig> deviceConfigCache = null;
>> +	
>> +	public PrimareBinding() {
>> +	}
>> +
>> +	public void activate() {
>> +		logger.debug("Activate");
>> +	}
>> +
>> +	public void deactivate() {
>> +		logger.debug("Deactivate");
>
> remove
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab/pull/2263/files#r30634708>.
>